File: src/code/fastpow/fastpow.java
class FastPow {
    static long calc(long a, long b, long MOD) {
        long base = a % MOD, res = 1;
        while (b != 0) {
            if ((b & 1) == 1) {
                res = res * base % MOD;
            }
            base = base * base % MOD;
            b >>>= 1;
        }
        return res % MOD;
    }

    static long calc(long a, long b) {
        int MOD = (int) 1e9 + 7;
        return calc(a, b, MOD);
    }
}

File: src/code/fastpow/fast_pow.rs
fn fast_pow(a: i64, mut n: i64, m: i64) -> i64 {
    let mut b = a % m;
    let mut res = 1;
    while n > 0 {
        if n & 1 == 1 {
            res = (res * b) % m;
        }
        b = (b * b) % m;
        n = n / 2;
    }
    return res % m;
}